quoridor-logic-server/src/main/java/com/rl/mpquoridor/models/game/GameManager.java

```diff
@@ -58,9 +61,13 @@ public GameResult run() {
                 logger.info("Illegal action " + e.getMessage());
                 continue;
             }
+            isGameEnded = (this.gameBoard.getWinner() != null);
+            List<Position> nextPlayerMoves = this.gameBoard.getCurrentPlayerMoves(this.playerPawn.get(currentPlayer));
+
             history.add(new HistoryRecord(currentPawn, action));
-            trigger(new TurnActionEvent(this.playerPawn.get(currentPlayer), action));
             this.players.add(this.players.poll()); // Move the current player to the end of the queue
+            trigger(new TurnActionEvent(this.playerPawn.get(currentPlayer), action,
```

Comment:
u cant return here action 
u need to return the new position of the player (u can step up the players)/ return the position of the wall